Iss223 -- deaths from injury

vpancini opened the pull request.

vpancini commented:
Data used for this metric are available on [Box](https://urbanorg.box.com/s/tulmxqirl015c4l1eivs2v68dwrjx7rx)

wcurrangroome commented on 04_health/exposure-to-trauma/final/exposure-to-trauma-2020.csv:
It looks like this is the output file that was generated from a prior run of the script when the file output name was different. This shouldn't be included in the commit that's merged into the base branch.

vpancini replied:
Oops. Thanks for flagging. Fixing for next PR.

wcurrangroome commented on 04_health/exposure-to-trauma/exposure-to-trauma-2020.Rmd:
Any reason for .Rmd over .qmd? If possible (it should be pretty easy), probably worthwhile to use .qmd over .Rmd at this point.

vpancini replied:
No. Have switched to .qmd.

wcurrangroome reviewed (changes requested):
Minor suggestions throughout, on the whole looks great! I think the primary outstanding item is to figure out and implement an approach to crosswalking a couple of geographies that don't align between CDC wonder data and our universe of counties. Can you check with Aaron and JP (and/or whomever the right folks are) about guidance on this? If nothing existing, the `interpolate_pw()` function from library(tidycensus) is likely a fine approach.

vpancini commented on 04_health/exposure-to-trauma/exposure-to-trauma-2020.Rmd:
```diff
@@ -0,0 +1,434 @@
+  + Hispanic Origin: All Origins
+  + Single Race 6: All Races
+4. Select year and month:
+  + Year/Month: 2016. 2017, 2018, 2019, 2020
```
Selecting `2016, 2017, 2018, 2019, 2020` creates a 5-year pool that is used as the data for 2020 only. (To get the data for 2019, the user would select the years `2015, 2016, 2017, 2018, 2019`. I will be more explicit in my instruction

vpancini commented on 04_health/exposure-to-trauma/exposure-to-trauma-2020.Rmd:
````diff
@@ -0,0 +1,434 @@
+
+```{r}
+# List all the files in the `bridged-race` folder
+paths <- list.files(here::here("04_health", "exposure-to-trauma", "data", "bridged-race"),
````
Data stored on Box will not be accessible to the public. Users will have to manually download the data. I will make the instructions for that section more clear.

vpancini commented on 04_health/exposure-to-trauma/exposure-to-trauma-2020.Rmd:
```diff
@@ -0,0 +1,434 @@
+library(readxl)
+library(writexl)
+
+library(rvest)
```
Good call. I always intend to do this and usually forget. Most of those packages were used in last year's update of the metric to read in the data from a different source.

vpancini commented on 04_health/exposure-to-trauma/exposure-to-trauma-2020.Rmd:
```diff
@@ -0,0 +1,434 @@
+)
+
+# Read in CHR data
+chr2022 <- readxl::read_excel(here::here("04_health", "exposure-to-trauma", "data", "chr2022.xlsx"),
```
Setting `guess_max = Inf` produced a different warning: `Warning: `guess_max` is a very large value, setting to `21474836` to avoid exhausting memoryNew names:` - but I still think this is better than the previous warning which was `Warning: Expecting logical in Q3172 / R3172C17: got 'x'New names:`.

vpancini commented on 04_health/exposure-to-trauma/exposure-to-trauma-2020.Rmd:
````diff
@@ -0,0 +1,434 @@
+This section reads in the Urban Institute's universe of counties, joins the CDC injury death data to those counties, and writes out the file.
+
+Read in Urban county file for 2014-2020
+```{r}
````
Done. Thanks for fixing that!

vpancini commented on 04_health/exposure-to-trauma/exposure-to-trauma-2020.Rmd:
```diff
@@ -0,0 +1,434 @@
+#     county_name, GEOID, year,
+#     rate_injury_death
+#   ) %>%
+#   reactable(
```
I needed to update a dependency for `library(reactable)` and then it worked.

vpancini commented on 04_health/exposure-to-trauma/exposure-to-trauma-2020.Rmd:
```diff
@@ -0,0 +1,434 @@
+library(rvest)
+library(glue)
+library(fs)
+# library(reactable)
```
I needed to update a dependency for `library(reactable)` and then it worked.

vpancini commented on 04_health/exposure-to-trauma/exposure-to-trauma-2020.Rmd:
````diff
@@ -0,0 +1,434 @@
+
+```
+
+Data for Bedford city, VA (FIPS code 51515) are only available for 2013 and earlier; for computational purposes, counts and population estimates for 2014 and later are set to zero. 
````
I moved the section that reads in Urban's universe of counties and joins the CDC WONDER onto it to the beginning. The missing death rates in this section aren't available because these counties no longer exist as such and have been included in the data only for consistency. I included more context explaining this and have removed them from the data.

vpancini commented on 04_health/exposure-to-trauma/exposure-to-trauma-2020.Rmd:
````diff
@@ -0,0 +1,434 @@
+
+
+### Visualizations of CDC WONDER data
+```{r}
````
I removed observations with missing population counts (the Urban universe of counties is missing population data for 2014) and changed the scale of the x axis to thousands.

vpancini commented on 04_health/exposure-to-trauma/exposure-to-trauma-2020.Rmd:
````diff
@@ -0,0 +1,434 @@
+### Compare values from CHR and CDC WONDER
+
+Visually inspecting the data shows that, for most observations, the rate produced by CHR matches the crude rate from CDC WONDER. There are 161 observations that do not match. These are observations where the rate was `unreliable` because the death count was less than 20. CHR produced the crude rate by dividing the death count by population for these counties and multiplying by 100,000. This compu ...
+```{r}
````
I printed the first 10 rows of data to show that these CHR observations have a death count value of less than 20. I then used a `stopifnot()` function to illustrate that this is true for all these observations. I was more explicit in my explanation that the CDC WONDER documentation states that they mark the rate as `unreliable` when the death count is less than 20, while CHR chose to calculate the rate for those observations anyway.

vpancini commented on 04_health/exposure-to-trauma/exposure-to-trauma-2020.Rmd:
```diff
@@ -0,0 +1,434 @@
+
+# Change all variables to numeric and replace suppressed or unreliable with missing
+deaths_14_20 <- deaths_14_20 %>%
+  # mutate_all(~ifelse(. %in% c("Suppressed", "Unreliable"), NA, .)) %>%
```
I agree the commented out approach is better. I initially went with the currently un-commented out approach because it replaced those text values with `NA` AND converted the rates and counts from character variables to numeric. With the commented out approach, I had to convert the variables in a separate step, so I initially chose not to use that approach. However, I agree it is clearer and have gone with the commented out approach instead.

vpancini commented on 04_health/exposure-to-trauma/exposure-to-trauma-2020.Rmd:
```diff
@@ -0,0 +1,434 @@
+  + Precision: 1 decimal places
+  + Data Access Timeout: 10 minutes
+  
+The specifications for the 2020 extract are saved [here](https://wonder.cdc.gov/controller/saved/D76/D377F651). Note that counties with fewer than ten deaths (for all the ICD-10 codes specified) are suppressed. Population figures are also suppressed when the population represents fewer than ten persons. Mortality rates are marked as "unreliable" when the death count is less than 20. Rates are mar ...
```
I've clarified these instructions

vpancini commented on 04_health/exposure-to-trauma/exposure-to-trauma-2020.Rmd:
```diff
@@ -0,0 +1,434 @@
+# List all the files in the `bridged-race` folder
+paths <- list.files(here::here("04_health", "exposure-to-trauma", "data", "bridged-race"),
+                    pattern = "[.]txt$", full.names = TRUE)
+
```
Included this check

vpancini commented on 04_health/exposure-to-trauma/exposure-to-trauma-2020.Rmd:
````diff
@@ -0,0 +1,434 @@
+```
+
+Chugach Census Area, AK and Copper River Census Area, AK are still missing mortality data from CDC. In early 2019, Valdez-Cordova Census Area, AK was abolished and replaced by the Chugach Census Area and the Copper River Census Area. 
+    todo(): not sure if we already have a way to crosswalk this
````
I implemented this for the age-adjusted rate. My understanding is that this approach won't work for interpolating the confidence interval or standard error because the population is smaller in each of Chugach Census Area and Copper River Census Area than in Valdez-Cordova Census Area.

wcurrangroome replied:
Thanks for trying this! I should have thought this through more carefully before making this suggestion; I don't think this is working correctly. I think the fundamental challenge is that the metric we're reporting (death rates) is standardized by population. So when we interpolate population (count) and deaths (count) and then calculate the deaths rates for each of the new census areas... we're just back at the original death rate we started with. My suggestion would be to just set the death rate for these new census areas to be the same and to set the data quality flag for these observations as low since I'm not sure what else we can do here. But I'll link us up with Aaron in a thread to double check.

vpancini commented on 04_health/exposure-to-trauma/exposure-to-trauma-2020.Rmd:
```diff
@@ -0,0 +1,434 @@
+                          )
+
+# Check for observations that did not match
+test <- anti_join(all_counties, deaths, by = c("GEOID", "year"))
```
Included where necessary. Also loaded the `tidylog` package at the beginning which helps with some of this.

vpancini commented on 04_health/exposure-to-trauma/exposure-to-trauma-2020.Rmd:
```diff
@@ -0,0 +1,434 @@
+# Check for observations that did not match
+test <- anti_join(all_counties, deaths, by = c("GEOID", "year"))
+
+# Remove obsolete variables
```
No, and this whole step was rearranged.

vpancini commented on 04_health/exposure-to-trauma/exposure-to-trauma-2020.Rmd:
```diff
@@ -0,0 +1,434 @@
+# Remove obsolete variables
+final_data <- final_data %>%
+  select(-GEOID)
+
```
Added where necessary

vpancini commented on .gitignore:
```diff
@@ -1,3 +1,14 @@
+vincent
```
It doesn't, I was just following existing convention that I noticed in the file (other people seemed to write their name above the files they added to the `.gitgnore` document. Have deleted.

vpancini commented on 04_health/exposure-to-trauma/exposure-to-trauma-2020.Rmd:
```diff
@@ -0,0 +1,434 @@
+    rate_injury_death_se = Age.Adjusted.Rate.Standard.Error
+  )
+
+# Remove empty rows resulting from deleting the `Notes` column, and remove the total row
```
Yes it does. I have clarified.

vpancini commented on .gitignore:
```diff
@@ -1,3 +1,14 @@
+04_health/exposure-to-trauma/cdc2021_single_race.txt
+04_health/exposure-to-trauma/chr2023.xlsx
+
  kevin
```
`04_health/exposure-to-trauma/chr2023.xlsx` is still relevant but have removed the other two lines.

vpancini commented:
I have addressed all reviewer comments and pushed the updated files.

vpancini commented:
> * There's one chunk with a testthat assertion that isn't fully built out (maybe it still works as-is?). If you search for "Check that these observations are equal" you should find it; the full code block might look like: `testthat::expect_equal( not_missing$rate_injury_death_chr, not_missing$rate_injury_death_crude)`
> * At "4.3 Identify counties in CDC Wonder but not in Urban universe": I'm seeing 36 observations, but the text refers to 50. Can you double-check this? I'm running things interactively, so I might be getting a different number because I've changed a variable relative to the value it would have at this point in the execution of the script (though I don't think that's the issue).
> * Per discussion, address the two AK census areas.
> * Then, run one last time top-to-bottom and this should be good to go!

- Did you maybe misread this line? This is what I'm seeing in my code:
`# Check that these observations are equal to the CHR observations
testthat::expect_equal(
  not_missing$rate_injury_death_chr,
  not_missing$rate_injury_death_crude
)`
- Good catch. There were 50 observations initially, but after addressing the counties in Urban universe but not in CDC WONDER in section 4.2, this drops to 36. I have updated the text accordingly.
- Addressed the two AK census areas. They now have the same rate and a quality value of `3` to indicate low quality
- Ran everything one chunk at a time to check and then rendered the file!
